### ISSUE TITLE
Can't generate or upload anonymous stats

### ISSUE THREAD

**Issue by fraschizzato** [bp2008/biupdatehelper#5]
I don't get the json and the anonymous stats. In the log I have:

```
28/08/2020 20:33:38	System.Exception: Unable to log in to server
   in BiUpdateHelper.CameraWebInterfaceLinker.GetSecureAuthenticatedSession(WebClient wc, Boolean& isAdmin, String user, String pass)
   in BiUpdateHelper.CameraWebInterfaceLinker.GenerateWebInterfaceLinkDocument(String outPath)
28/08/2020 20:34:10	Unable to generate anonymous performance data record.
System.ArgumentNullException: Il valore non può essere null.
Nome parametro: key
   in System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   in System.Collections.Generic.SortedList`2.Add(TKey key, TValue value)
   in BlueIrisRegistryReader.BlueIrisConfiguration.Load() in C:\Users\brpea\Source\Repos\BlueIrisRegistryReader\BlueIrisRegistryReader\BlueIrisConfiguration.cs:riga 47
   in BiUpdateHelper.PerformanceDataCollector.GetPerfDataRecord()
   in BiUpdateHelper.PerformanceDataCollector.PerfDataReportBackground()
```

Thanks

**Comment by bp2008** [comment]
Hi.

Which version of BiUpdateHelper are you using?

The log suggests two problems.  One, the app failed to log in to your Blue Iris web server.  Unfortunately we don't know why it failed.  Can you use Blue Iris's web interface or mobile apps normally?

The second error suggests that one of your camera instances in Blue Iris does not have a short name specified.  This should not be possible normally, so I am thinking maybe there is some registry corruption or something else strange going on.  Can you open `regedit` and go to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Perspective Software\Blue Iris\Cameras` and screenshot the contents of that?

For example here's mine:

![image](https://user-images.githubusercontent.com/5639911/91610429-6cbc3180-e936-11ea-89d7-9e227f8d82c3.png)

**Comment by fraschizzato** [comment]
Version is 1.9
Both App and UI3 are working well. Here my registry entries:
![Cam1](https://user-images.githubusercontent.com/4145029/91611009-96974780-e97a-11ea-85db-e1e1ac044e6f.png)
![Cam2](https://user-images.githubusercontent.com/4145029/91611013-98610b00-e97a-11ea-8d47-736100c12ea6.png)

**Comment by bp2008** [comment]
Ok, well unfortunately nothing obvious is wrong there.  Here's a slightly updated release that should add more information to the log for both of the errors you showed.

[BiUpdateHelper_1.9.0.1.zip](https://github.com/bp2008/biupdatehelper/files/5144059/BiUpdateHelper_1.9.0.1.zip)

Please run that and see what the log says.

**Comment by bp2008** [comment]
That should tell us exactly what the server's response was when login failed, and also give us the name of one of those registry folders that didn't have a valid short name.  (hopefully)

**Comment by fraschizzato** [comment]
The second error is solved, the camera was (ironically) "test". The short name field was not empty but filled with the name "Camera", changing it solved the problem. 

The software seems to work without issues now (there is only an "unable to authenticate" for all cameras in the snapshot link of camera links), but the other log entry is still there:

```
29/08/2020 04:14:38	Error dealing with web server.
System.Exception: Unable to log in to server. Login response was: {"result":"fail","session":"29b6492042f43c401b8d45d32dcf613d","data":{"auth-exempt":false,"reason":"account disabled"}}
   in BiUpdateHelper.CameraWebInterfaceLinker.GetSecureAuthenticatedSession(WebClient wc, Boolean& isAdmin, String user, String pass)
   in BiUpdateHelper.PerformanceDataCollector.GetPerfDataRecord()
```

Thanks

**Comment by fraschizzato** [comment]
Also solved the other message, and cam links issue. The "biupdatehelperuser" was disabled in BI.
Thanks

**Comment by bp2008** [comment]
Interesting.  I wonder why that failed.  Oh well.  I'll see if I can add some logic to make sure the user is enabled.  I'm surprised it isn't already ...

**Comment by fraschizzato** [comment]
Probably both the issues comes from my configuration restore when I had to update to Win10. For the update i have made a fresh install and a BI config restore. Probably the camera short name issue comes from that. As I didn't initially restored biupdatehelper, probably the user was disabled from mine during a full configuration check...   
Thanks

**Comment by bp2008** [comment]
I've just released version 1.9.1 which includes the logging additions from that 1.9.0.1 build I linked earlier, plus a fix for the "biupdatehelperuser" user being disabled (it just gets re-enabled every time it is used, which is okay because it also gets a new complex random password each time it is used).

**Comment by bp2008** [comment]
Thanks for your report and your help dealing with these cases! :)
